plugins/ccgx/fu-ccgx-hpi.c

```diff
@@ -0,0 +1,1622 @@
+			g_prefix_error (error, "fu_ccgx_hpi_read_event_reg - read response reg error:");
+			return FALSE;
+		}
+
+		/* byte 1 is reserved and should read as zero */
+		data_buffer[1] = 0;
+		memcpy ((guint8*)event, data_buffer, 4);
+		if (event->event_length != 0) {
+			if (!fu_ccgx_hpi_reg_read (device, &handle->i2c_handle, handle->hpi_addr_byte,
+					fu_ccgx_hpi_reg_addr_gen (section, HPI_REG_PART_PDDATA_READ, 4), event->event_data,
+					event->event_length, error)) {
+				g_prefix_error (error, "fu_ccgx_hpi_read_event_reg - read event data error:");
+				return FALSE;
+			}
+		}
+	}
+	else
+	{
+		if (!fu_ccgx_hpi_reg_read (device, &handle->i2c_handle,handle->hpi_addr_byte, CY_PD_REG_RESPONSE_ADDR ,data_buffer, 2, error)) {
+			g_prefix_error (error, "fu_ccgx_hpi_read_event_reg - read response reg error:");
+			return FALSE;
+		}
+		event->event_code   = data_buffer[0];
+		event->event_length = data_buffer[1];
+
+		if (event->event_length != 0) {
+			/* read the data memory */
+			if (!fu_ccgx_hpi_reg_read (device, &handle->i2c_handle, handle->hpi_addr_byte,
+				       CY_PD_REG_BOOTDATA_MEMEORY_ADDR,event->event_data, event->event_length,error)) {
+				g_prefix_error (error, "fu_ccgx_hpi_read_event_reg - read event data error:");
+				return FALSE;
+			}
+		}
+	}
+
+	return fu_ccgx_hpi_clear_intr (device, handle, section, error);
+}
+
+static gint8
+fu_ccgx_hpi_app_read_intr_reg (FuDevice *device, CyHPIHandle *handle, HPIRegSection section, HPIEvent *event_array, GError **error)
+{
+	gint8 event_count = -1;
+	guint8 intr_reg =0;
+	guint8 i;
+
+	g_return_val_if_fail (handle != NULL, -1);
+
+	if (!fu_ccgx_hpi_reg_read (device, &handle->i2c_handle,handle->hpi_addr_byte,
+		fu_ccgx_hpi_reg_addr_gen (HPI_REG_SECTION_DEV, HPI_REG_PART_REG, HPI_DEV_REG_INTR_ADDR), &intr_reg, 1, error)) {
+		g_prefix_error (error, "read_intr_reg  read intr reg error:");
+		return -1;
+	}
+
+	event_count = 0;
+	/* device section will not come here. */
+	for (i = 0; i <= handle->num_of_ports; i++) {
+		/* check wheteher this section is needed */
+		if ((section == i) || (section == HPI_REG_SECTION_ALL)) {
+			/* check whether this section has any event/response */
+			if ((1 << i) & intr_reg) {
+				if (!fu_ccgx_hpi_read_event_reg (device, handle, section, &event_array[i], error)) {
+					g_prefix_error (error, "read_intr_reg  read event error:");
+					return -1;
+				}
+				event_count++;
+			}
+		}
+	}
+
+	return event_count;
+}
+
+static gboolean
+fu_ccgx_hpi_wait_for_event (FuDevice *device, CyHPIHandle *handle, HPIRegSection section, HPIEvent *event_array, guint32 timeout_ms, GError **error)
+{
+	guint64 elapsed_time_ms = 0;
+	g_autoptr(GTimer) start_time = g_timer_new ();
+	gint8 event_count;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	g_return_val_if_fail (event_array != NULL, FALSE);
+
+	do {
+			event_count = fu_ccgx_hpi_app_read_intr_reg (device,handle, section, event_array,error);
```

Comment:
if you're using the `error` you need to return FALSE here -- or you can use error_local and just print the error

Reply:
I want to use error_local here. 
Could you check my code snippet attached?

![image](https://user-images.githubusercontent.com/32348354/77037903-be5ea880-69f5-11ea-9b0b-a512b70bdb1a.png)

Reply:
I fixed it using error_local